src/old-components/Guilds/ActionsBuilder/SupportedActions/SetPermissions/SetPermissionsEditor.tsx

```diff
@@ -0,0 +1,313 @@
+  // and the ANY_FUNC_SIGNATURE that is switched when in AssetTransfer component
+  const [customFunctionName, setCustomFunctionName] = useState(
+    parsedData?.functionName
+  );
+  const handleCustomFunctionSignature = (value: string) => {
+    setCustomFunctionName(value);
+    setFunctionSignature(value);
+  };
+  useEffect(() => {
+    if (activeTab === 0) setFunctionSignature(ANY_FUNC_SIGNATURE);
+    if (activeTab === 1) setFunctionSignature(customFunctionName);
+  }, [activeTab]);
+
+  //
+  // Amount related functions
+  //
+
+  // valueAllowed
+  const setAmount = (valueAllowed: BigNumber) => {
+    updateCall({
+      ...decodedCall,
+      args: {
+        ...decodedCall.args,
+        valueAllowed: [valueAllowed],
+      },
+    });
+  };
+
+  const [customAmountValue, setCustomAmountValue] = useState(
+    parsedData?.valueAllowed[0]
+  );
+  // This function was implemented to avoid the amount input to
+  // change to MAX_UINT toggling to "Max value"
+  const handleTokenAmountInputChange = (e: BigNumber) => {
+    setAmount(e);
+    setCustomAmountValue(e);
+  };
+
+  const [maxValueToggled, setMaxValueToggled] = useState(false);
+  const bigNumberMaxUINT = BigNumber.from(MAX_UINT);
+  const handleToggleMaxValueChange = () => {
+    if (!maxValueToggled) setAmount(bigNumberMaxUINT);
+    else setAmount(customAmountValue);
+    setMaxValueToggled(!maxValueToggled);
+  };
+
+  //
+  // To-Address related functions
+  //
+
+  const [customToAddress, setCustomToAddress] = useState(parsedData?.to[0]);
+  const [anyAddressToggled, setAnyAddressToggled] = useState(
+    parsedData?.to[0] === ANY_ADDRESS
+  );
+  // to address
+  const setToAddress = (to: string) => {
+    updateCall({
+      ...decodedCall,
+      args: {
+        ...decodedCall.args,
+        to: [to],
+      },
+    });
+  };
+
+  const handleToggleAnyAddressChange = () => {
+    if (!anyAddressToggled) setToAddress(ANY_ADDRESS);
+    else setToAddress(customToAddress);
+    setAnyAddressToggled(!anyAddressToggled);
+  };
+
+  const handleCustomAddress = (value: string) => {
+    setCustomToAddress(value);
+    if (value === '') {
+      setToAddress(ANY_ADDRESS);
+    } else {
+      setToAddress(value);
+    }
+  };
+
+  // If the 'to' address is ANY_ADDRESS, set customAmount to '', to
+  // show the address input empty, instead of the long 0xAaaAaaa address
+  useEffect(() => {
+    if (parsedData?.to[0] === ANY_ADDRESS) handleCustomAddress('');
+  }, []);
+
+  return (
+    <div>
+      <DetailWrapper>
+        <TabButton
+          aria-label="assets transfer tab"
+          active={activeTab === 0}
+          onClick={() => setActiveTab(0)}
+        >
+          Assets transfer
+        </TabButton>
+        <TabButton
+          aria-label="functions call tab"
+          active={activeTab === 1}
+          onClick={() => setActiveTab(1)}
+        >
+          Functions call
+        </TabButton>
+      </DetailWrapper>
+      {activeTab === 0 && (

```

Comment:
Do you think it would be difficult to make the tab its own component that we can pass an array of objects to with the titles and children to render?